hardware_interface/src/resource_manager.cpp

```diff
@@ -756,50 +793,83 @@ ResourceManager::ResourceManager(
 }
 
 // CM API: Called in "callback/slow"-thread
-void ResourceManager::load_urdf(
-  const std::string & urdf, bool validate_interfaces, bool load_and_initialize_components)
+bool ResourceManager::load_and_initialize_components(const std::string & urdf)
```

Comment:
This is nice, but I'm wondering how we will deal with the components in the gazebo?

https://github.com/ros-controls/gazebo_ros2_control/blob/ef4caaac3335ee1d7bd51087d062f6a8c5989dc2/gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp#L360

Gazebo seems to load it's components and setup the resource_manager, so in this case, when `are_components_initialized` is called, it needs to return `true`. If not, the simulation will fail to start. How can we deal with that situation here?

Reply:
If that workflow is used, then we don't check this in CM. We are then using different constructors in CM. 

Or am I wrong? Should we open PR for Gazebo too to see what is happening there?

Reply:
To be honest, I don't see that these methods collide. Or am I missing something? Gazebo also calls in line 292 `load_urdf` itself.

Reply:
So we will definitively break gazebo since it is using the public interface, anyway...

Reply:
If this is the case, I would recommend to already open a PR on the gazebo side fixing the same, so we can merge them together

Reply:
Do we have a link for that PR?

Reply:
This part of the code need to be modified : https://github.com/ros-controls/gazebo_ros2_control/blob/ef4caaac3335ee1d7bd51087d062f6a8c5989dc2/gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp#L291-L297